src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java

```diff
@@ -1,7 +1,237 @@
+                        Instant.now()
+                )
+        );
+
+        //WHEN
+        ResultActions editResultActions = mockMvc.perform(
+                patch("/api/wallet")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(Objects.requireNonNull(objectToJson(editWalletDto)))
+        );
+
+        //THEN
+        editResultActions.andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void itShouldReturnBadRequestWhenNameIsEmpty() throws Exception {
+        //GIVEN
+        TestEditWalletDto editWalletDto = returnTestEditWalletDtoWithEmptyName();
+        given(walletService.updateWallet(Mockito.any())).willReturn(
+                new WalletDTO(
+                        Mockito.anyLong(),
+                        Mockito.anyString(),
+                        Instant.now()
+                )
+        );
+
+        //WHEN
+        ResultActions editResultActions = mockMvc.perform(
+                patch("/api/wallet")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(Objects.requireNonNull(objectToJson(editWalletDto)))
+        );
+
+        //THEN
+        editResultActions.andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void itShouldReturnBadRequestWhenNameContainsWrongLetters() throws Exception {
+        //GIVEN
+        TestEditWalletDto editWalletDto = returnTestEditWalletDtoWithWrongLetterInName();
+        given(walletService.updateWallet(Mockito.any())).willReturn(
+                new WalletDTO(
+                        Mockito.anyLong(),
+                        Mockito.anyString(),
+                        Instant.now()
+                )
+        );
+
+        //WHEN
+        ResultActions editResultActions = mockMvc.perform(
+                patch("/api/wallet")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(Objects.requireNonNull(objectToJson(editWalletDto)))
+        );
+
+        //THEN
+        editResultActions.andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void itShouldReturnBadRequestWhenIdIsEmpty() throws Exception {
+        //GIVEN
+        TestEditWalletDto editWalletDto = returnTestEditWalletDtoWithNoId();
+        given(walletService.updateWallet(Mockito.any())).willReturn(
+                new WalletDTO(
+                        Mockito.anyLong(),
+                        Mockito.anyString(),
+                        Instant.now()
+                )
+        );
+
+        //WHEN
+        ResultActions editResultActions = mockMvc.perform(
+                patch("/api/wallet")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(Objects.requireNonNull(objectToJson(editWalletDto)))
+        );
+
+        //THEN
+        editResultActions.andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void itShouldReturnBadRequestWhenIdIsZero() throws Exception {
```

Comment:
Test should fail when `id` is lower than '1'. Value is lower than `1` mean it's `0` or even negative value.

Reply:
I propose to add one more test, where ID with negative value will be tested. Is that okay?

Reply:
Decided to add new test to check when ID will be negative.